engine/class_modules/sc_death_knight.cpp

```diff
@@ -8646,11 +8746,15 @@ void death_knight_t::init_spells()
 
   // Shadowlands specific - Commented out = NYI
 
+  // Covenants
+  covenant.swarming_mist = find_covenant_spell( "Swarming Mist" );
+  covenant.swarming_mist_energize = covenant.swarming_mist->ok() ? find_spell( 312546 ) : spell_data_t::not_found();
```

Comment:
If it's only used inside the swarming mist tick, you could just check it on the tick creation and store the value as a variable there.

Reply:
```
orig
    swarming_mist              Count=   5.5| 60.000sec  DPE= 8292| 0.00%  DPET=7035  DPR=  0  pDPS=  0
    swarming_mist_damage       Count=  48.9|  5.904sec  DPE=  932|19.41%  DPET=  0  DPR=  0  pDPS=152  Miss= 0.00%  Hit=   800|   730|   905  Crit=  1598|  1460|  1811|16.59%
    swarming_mist                : start=  5.5 refresh=  0.0 interval= 60.0 trigger= 60.0 duration=  7.9 uptime= 14.54%

refactor
    swarming_mist              Count=   5.5| 60.000sec  DPE= 8315| 0.00%  DPET=7055  DPR=  0  pDPS=  0
    swarming_mist_damage       Count=  49.1|  5.872sec  DPE=  933|19.44%  DPET=  0  DPR=  0  pDPS=152  Miss= 0.00%  Hit=   800|   730|   905  Crit=  1599|  1460|  1811|16.69%
    swarming_mist                : start=  5.5 refresh=  0.0 interval= 60.0 trigger= 60.0 duration=  7.9 uptime= 14.52%
```